blaze/utils.py

```diff
@@ -266,3 +272,20 @@ def parameter_space(*args):
     return tuple(product(*(
         arg if isinstance(arg, tuple) else (arg,) for arg in args
     )))
+
+
+def memoize_mutable(fn):
```

Comment:
@skrah can you test this implementation on an expression tree that uses a `blaze.Data` object that's referring to a large  array or dataframe, and check and see if the entire contents of the array / dataframe are serialized?  I want to make sure that we don't serialize or memoize GBs of data with this implementation.